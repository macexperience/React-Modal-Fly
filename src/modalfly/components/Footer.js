import { useState, useEffect } from 'react';
import ReactDOM from 'react-dom';
import './styles.css';

export function Footer(props) {
    const [footerRef, setFooterRef] = useState(null);

    //When component  mounts, set footer reference
    useEffect(() => {
        //Query all footer areas    
        const elements = document.getElementsByClassName('mf-footer-area');
        //Select the most recent addition, the div with no children
        let footerRef;
        for (let i = 0; i < elements.length; i++) {
            if (!elements[i].hasChildNodes()) footerRef = elements[i];
        }

        var zIndex = Math.max.apply(null, Array.prototype.map.call(document.querySelectorAll('*'), function(el) {
            return +el.style.zIndex;
          })) + 10;
        footerRef.style.zIndex = zIndex * 2;

        setFooterRef(footerRef);
    }, []);

    if (footerRef) {
        //Render the buttons in the mf-footer div ('footerRef')
        return ReactDOM.createPortal(props.children, footerRef);
    } else {
        return null;
    }
}