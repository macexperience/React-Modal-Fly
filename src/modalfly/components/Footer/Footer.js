import { useState, useEffect } from 'react';
import ReactDOM from 'react-dom';

export function Footer(props) {
    const [footerRef, setFooterRef] = useState(null);

    //When component  mounts, set footer reference
    useEffect(() => {
        //Query all footer areas    
        const elements = document.getElementsByClassName('mf-footer-area');
        //No do not set footer reference if elements is empty
        if (elements.length === 0) {
            return;
        }
        //Select the most recent addition, the div with no children
        let footerRef;
        for (let i = 0; i < elements.length; i++) {
            if (!elements[i].hasChildNodes()) footerRef = elements[i];
        }

        var zIndex = Math.max.apply(null, Array.prototype.map.call(document.querySelectorAll('*'), function(el) {
            return +el.style.zIndex;
          })) + 10;
        footerRef.style.zIndex = zIndex * 2;

        setFooterRef(footerRef);
    }, []);

    if (footerRef) {
        //Render the buttons in the mf-footer div ('footerRef')
        return ReactDOM.createPortal(props.children, footerRef);
    } else {
        return null;
    }
}