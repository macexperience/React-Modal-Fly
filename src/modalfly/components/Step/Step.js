import React from 'react';

export function Step(props) {
    return (
        <div title={props.title} className={props.className} style={props.style}>
            {props.children}
        </div>
    );
};