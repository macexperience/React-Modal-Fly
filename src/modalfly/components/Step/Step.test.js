import React from 'react';
import renderer from 'react-test-renderer';
import { Step } from './Step';
import { Footer } from '../Footer/Footer';

describe('<Step />', () => {
    it('renders and mounts with no children', () => {
        expect.assertions(1);
        const component = renderer.create(<Step />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders and mounts with children', () => {
        expect.assertions(1);
        const component = renderer.create(
            <Step>
                <h3>Step 1</h3>
                <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod voluptate dignissimos beatae, architecto, unde vero provident facilis recusandae laborum quam eligendi. Error at placeat eveniet labore blanditiis beatae, maiores adipisci.</p>
                <Footer>
                    <button type='button'>Next</button>
                </Footer>
            </Step>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});