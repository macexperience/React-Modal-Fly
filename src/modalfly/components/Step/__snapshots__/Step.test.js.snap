// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> renders and mounts with children 1`] = `
<div>
  <h3>
    Step 1
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod voluptate dignissimos beatae, architecto, unde vero provident facilis recusandae laborum quam eligendi. Error at placeat eveniet labore blanditiis beatae, maiores adipisci.
  </p>
</div>
`;

exports[`<Step /> renders and mounts with no children 1`] = `<div />`;
